File: cyanBat/src/main/java/at/msmiech/cyanbat/interfaces/GameObject.java
package at.msmiech.cyanbat.interfaces;

import java.util.List;

import at.grueneis.game.framework.Graphics;
import at.grueneis.game.framework.Input.TouchEvent;
import at.msmiech.cyanbat.CyanBatGame;

public interface GameObject {
	public static final String TAG = CyanBatGame.TAG;
	public static final boolean DEBUG = CyanBatGame.DEBUG;
	public void update(float deltaTime, List<TouchEvent> touchEvents);
	public void draw(Graphics g);
}

File: app/src/main/kotlin/at/smiech/cyanbat/gameobject/GameObject.kt
package at.smiech.cyanbat.gameobject

import at.grueneis.game.framework.Graphics
import at.grueneis.game.framework.Input.TouchEvent
import at.grueneis.game.framework.math.Rect

/**
 * A common in-game-object that can be updated and drawn on screen
 *
 * @author mart1n8891
 */
interface GameObject {
    val rectangle: Rect

    fun update(deltaTime: Float, touchEvents: List<TouchEvent>)

    fun draw(g: Graphics)

    fun isScheduledForRemoval(): Boolean
}
